### PR TITLE
PR for Release 1.1.0 "Wingardium Leviosa"

### PR DESCRIPTION
**Pull request by ggabernet**
- Added log parsing
- Added fastQC and multiQC
- Added downstream repertoire analysis
- Added clonal analysis
- Added handling of index files

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If necessary, also make a PR on the [nf-core/bcellmagic branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/bcellmagic)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/bcellmagic/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,12 +9,12 @@ matrix:
...
 before_install:
   # PRs to master are only ok if coming from dev branch
-  - '[ $TRAVIS_PULL_REQUEST = "false" ] || [ $TRAVIS_BRANCH != "master" ] || ([ $TRAVIS_PULL_REQUEST_SLUG = $TRAVIS_REPO_SLUG ] && [ $TRAVIS_PULL_REQUEST_BRANCH = "dev" ])'
+  - '[ $TRAVIS_PULL_REQUEST = "false" ] || [ $TRAVIS_BRANCH != "master" ] || ([ $TRAVIS_PULL_REQUEST_SLUG = $TRAVIS_REPO_SLUG ] && ([ $TRAVIS_PULL_REQUEST_BRANCH = "dev" ] || [ $TRAVIS_PULL_REQUEST_BRANCH = "patch" ]))'
   # Pull the docker image first so the test doesn't wait for this
   - docker pull nfcore/bcellmagic:dev
   # Fake the tag locally so that the pipeline runs properly
   # Looks weird when this is :dev to :dev, but makes sense when testing code for a release (:dev to :1.0.1)
-  - docker tag nfcore/bcellmagic:dev nfcore/bcellmagic:1.0.0
+  - docker tag nfcore/bcellmagic:dev nfcore/bcellmagic:dev
```
Comment (apeltzer) [inline .travis.yml]:
```suggestion
  - docker tag nfcore/bcellmagic:dev nfcore/bcellmagic:1.1.0
```

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,8 @@
-FROM nfcore/base
+FROM nfcore/base:1.7
 LABEL authors="Gisela Gabernet, Simon Heumos, Alexander Peltzer" \
       description="Docker image containing all requirements for nf-core/bcellmagic pipeline"
 
 COPY environment.yml /
 RUN conda env create -f /environment.yml && conda clean -a
-ENV PATH /opt/conda/envs/nf-core-bcellmagic-1.0.0/bin:$PATH
-RUN ln -s /opt/conda/envs/nf-core-bcellmagic-1.0.0/bin/vsearch /opt/conda/envs/nf-core-bcellmagic-1.0.0/bin/usearch
+ENV PATH /opt/conda/envs/nf-core-bcellmagic-1.0.0dev/bin:$PATH
```
Comment (apeltzer) [inline Dockerfile]:
```suggestion
ENV PATH /opt/conda/envs/nf-core-bcellmagic-1.1.0/bin:$PATH
```

File: Dockerfile [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+RUN ln -s /opt/conda/envs/nf-core-bcellmagic-1.0.0dev/bin/vsearch /opt/conda/envs/nf-core-bcellmagic-1.0.0dev/bin/usearch
```
Comment (apeltzer) [inline Dockerfile]:
```suggestion
RUN ln -s /opt/conda/envs/nf-core-bcellmagic-1.1.0/bin/vsearch /opt/conda/envs/nf-core-bcellmagic-1.1.0/bin/usearch
```

File: README.md [hunk 4]
```diff
@@ -1,24 +1,21 @@
-# nf-core/bcellmagic
-
-A pipeline to analyze B-cell repertoires.
+# ![nf-core/bcellmagic](docs/images/Bcellmagic_logo.png)
 
 [![Build Status](https://travis-ci.org/nf-core/bcellmagic.svg?branch=master)](https://travis-ci.org/nf-core/bcellmagic)
```
Comment (apeltzer) [inline README.md]:
```suggestion
[![Build Status](https://travis-ci.com/nf-core/bcellmagic.svg?branch=master)](https://travis-ci.com/nf-core/bcellmagic)
```

File: nextflow.config [hunk 5]
```diff
@@ -54,7 +53,7 @@ params {
 
 // Container slug. Stable releases should specify release tag!
 // Developmental code should specify :dev
-process.container = 'nfcore/bcellmagic:1.0.0'
+process.container = 'nfcore/bcellmagic:dev'
```
Comment (apeltzer) [inline nextflow.config]:
```suggestion
process.container = 'nfcore/bcellmagic:1.1.0'
```

File: nextflow.config [hunk 6]
```diff
@@ -106,10 +101,10 @@ manifest {
   name = 'nf-core/bcellmagic'
   author = 'Gisela Gabernet, Simon Heumos, Alexander Peltzer'
   homePage = 'https://github.com/nf-core/bcellmagic'
-  description = 'B/T cell repertoire analysis pipeline with immcantation framework. WIP, currently requires a bunch of changes first.'
+  description = 'B cell repertoire analysis pipeline with immcantation framework.'
   mainScript = 'main.nf'
-  nextflowVersion = '>=0.32.0'
-  version = '1.0.0'
+  nextflowVersion = '>=19.04.0'
+  version = '1.0.0dev'
```
Comment (apeltzer) [inline nextflow.config]:
```suggestion
  version = '1.1.0'
```

File: main.nf [hunk 7]
```diff
@@ -36,8 +36,9 @@ def helpMessage() {
     Define clones:
       --set_cluster_threshold       Set this parameter to allow manual hamming distance threshold for cell cluster definition.
       --cluster_threshold           Once set_cluster_threshold is true, set cluster_threshold value (float).
-      --define_clones_only          If set, expects tables as produced by change-O as input, tab delimited. Only performs clonal and germline assignment.
-      --changeo_tsv                 Once define_clones_only is specified, set the path to the change-O tsv file.
+    
+    Index file:
+      --index_file                  If Unique molecular Identifier is written in a separate index file, merge it to R1 reads.
```
Comment (apeltzer) [inline main.nf]:
```suggestion
      --index_file                  If a unique molecular identifier (UMI) is written in a separate index file, merge it to R1 reads.
```

File: main.nf [hunk 8]
```diff
@@ -733,11 +703,9 @@ process assign_clones{
     tag "${id}" 
     publishDir "${params.outdir}/define_clones/$id", mode: 'copy',
         saveAs: {filename ->
-            if (filename.indexOf(".fasta") > 0) "fasta/$filename"
-            else if (filename.indexOf(".log") > 0) null
-            else if (filename.indexOf("table.tab") > 0) "info/$filename"
-            else if (filename.indexOf(".tab") > 0) "table/$filename"
-            else if (filename == "command_log.txt") "$filename"
+            if (filename.indexOf("table.tab") > 0) "$filename"
+            else if (filename.indexOf("clone-pass.tab") > 0) "$filename"
```
Comment (Leon-Bichmann) [inline main.nf]: A lot of different cases of if statements in your code, I wonder if that could be done in an easier way. But if it works great! ;)

File: main.nf [hunk 9]
```diff
@@ -749,7 +717,7 @@ process assign_clones{
     set file("${geno.baseName}_clone-pass.tab"), file("$geno_fasta"), val("$id") into ch_for_germlines
     file "${geno.baseName}_clone-pass.tab"
```
Comment (Leon-Bichmann) [inline main.nf]: You define a lot of output files but don't pass them into channels, I wonder if it is necessary to define them at all.

File: main.nf [hunk 10]
```diff
@@ -110,46 +111,39 @@ if (params.set_cluster_threshold){
...
+    ch_read_files_for_merge_r1_umi_index = Channel.from(file_meta)
+            .splitCsv(header: true, sep:'\t')
+            .map { col -> tuple("${col.ID}", "${col.Source}", "${col.Treatment}","${col.Extraction_time}","${col.Population}", file("${col.R1}", checkifExists: true),file("${col.R2}", checkifExists: true), file("${col.I1}", checkifExists: true))}
+            .dump()
+    ch_read_files_for_merge_r1_umi = Channel.empty()
+} else {
+    ch_read_files_for_merge_r1_umi = Channel.from(file_meta)
+            .splitCsv(header: true, sep:'\t')
+            .map { col -> tuple("${col.ID}", "${col.Source}", "${col.Treatment}","${col.Extraction_time}","${col.Population}", file("${col.R1}", checkifExists: true), file("${col.R2}", checkifExists: true))}
+            .dump()
```
Comment (Leon-Bichmann) [inline main.nf]: Is dump() necessary for this?

File: main.nf [hunk 11]
```diff
@@ -166,7 +160,6 @@ summary['Max Time']     = params.max_time
 summary['IGDB Path']    = params.igblast_base
```
Comment (Leon-Bichmann) [inline main.nf]: You could print some more of the workflow options here,  right now you print very few.

**Pull request merged**